src/offset/local/mod.rs

```diff
@@ -137,7 +137,149 @@ impl TimeZone for Local {
 mod tests {
     use super::Local;
     use crate::offset::TimeZone;
-    use crate::Datelike;
+    use crate::{Datelike, Duration, NaiveDate};
+
+    use std::{path, process};
+
+    #[cfg(unix)]
+    fn verify_against_date_command_local(
+        path: &'static str,
+        year: i32,
+        month: u32,
+        day: u32,
+        hour: u32,
+    ) {
+        let output = process::Command::new(path)
+            .arg("-d")
+            .arg(format!("{year}-{month:02}-{day:02} {hour:02}:05:01"))
+            .arg("+%Y-%m-%d %H:%M:%S %:z")
+            .output()
+            .unwrap();
+
+        let date_command_str = String::from_utf8(output.stdout).unwrap();
+
+        // The below would be preferred. At this stage neither earliest() or latest()
+        // seems to be consistent with the output of the `date` command, so we simply
+        // compare both.
+        // let local = Local
+        //     .from_local_datetime(&NaiveDate::from_ymd(year, month, day).and_hms(hour, 5, 1))
+        //     // looks like the "date" command always returns a given time when it is ambiguous
+        //     .earliest();
+
+        // if let Some(local) = local {
+        //     assert_eq!(format!("{}\n", local), date_command_str);
+        // } else {
+        //     // we are in a "Spring forward gap" due to DST, and so date also returns ""
+        //     assert_eq!("", date_command_str);
+        // }
+
+        match Local.from_local_datetime(&NaiveDate::from_ymd(year, month, day).and_hms(hour, 5, 1))
+        {
+            crate::LocalResult::Ambiguous(a, b) => {
+                assert!(
+                    format!("{}\n", a) == date_command_str
+                        || format!("{}\n", b) == date_command_str
+                )
+            }
+            crate::LocalResult::Single(a) => {
+                assert_eq!(format!("{}\n", a), date_command_str);
+            }
+            crate::LocalResult::None => {
+                assert_eq!("", date_command_str);
+            }
+        }
+    }
+
+    #[test]
+    #[cfg(unix)]
+    fn try_verify_against_date_command() {
+        // #TODO: investigate /bin/date command behaviour on macOS
+        // avoid running this on macOS, temporarily
+        // for date_path in ["/usr/bin/date", "/bin/date"] {
+        for date_path in ["/usr/bin/date"] {
+            if path::Path::new(date_path).exists() {
```

Comment:
Can you make this use an early continue? That would help reduce rightward drift.

Reply:
good call, the code was a bit gnarly, I've made some improvements by using NaiveDateTime instead

Reply:
I removed the loop across date commands as I tried this out on macOS, but I couldn't find the same command line arguments and so wasn't able the replicate the behavior desired to find the local timezone